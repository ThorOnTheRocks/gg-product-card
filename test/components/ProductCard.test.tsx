import React from 'react';
import ProductCard from '../../src/components';
import renderer from 'react-test-renderer';
import { product1 } from '../data/products';

describe('ProductCard', () => {
  test('it should render the component correctly', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {() => <h1>Product Card</h1>}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('it should increase the counter', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {({ count, handleCounter }) => (
          <>
            <h1>Product Card</h1>
            <span>{count}</span>
            <button onClick={() => handleCounter(1)}>+</button>
          </>
        )}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
