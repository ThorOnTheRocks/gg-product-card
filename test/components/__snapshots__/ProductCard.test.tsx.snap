// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard it should increase the counter 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`ProductCard it should render the component correctly 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;
